src/api/student.js

```diff
@@ -5,10 +5,10 @@ import xlsx from 'xlsx';
 let app = express();
 
 app.get('/', (req, res) => {
-    let student = StudentStore.getStudentById(req.query.id);
+    let student = StudentStore.getStudentByRFID(parseInt(req.query.rfid));
     let regex = new RegExp("^[a-zA-Z0-9]+$");
 
-    if(!regex.test(req.query.id)){
+    if(!regex.test(req.query.rfid)){
```

Comment:
Why are we validating what's on the RFID cards? We don't require it to be only alphanumerics. Plus, that `parseInt` above will only work if `req.query.rfid` is already base10, or starts with a `0x` (hexadecimal) or `0` (octal or decimal).

I think it's safe to just remove any sort of validation on the RFID value; just assume it is correct..